From hohndel@linux.intel.com  Mon Oct 29 22:57:40 2007
From: Dirk Hohndel <hohndel@linux.intel.com>
Date: Mon, 29 Oct 2007 06:28:17 -0700
Subject: PCI: pciserial_resume_one ignored return value of pci_enable_device
To: Greg KH <gregkh@suse.de>, linux-kernel@vger.kernel.org
Message-ID: <20071029132817.GA20449@bigserver.hohndel.org>
Content-Disposition: inline

From: Dirk Hohndel <hohndel@linux.intel.com>

[PATCH] pciserial_resume_one ignored return value of pci_enable_device

Signed-off-by: Dirk Hohndel <hohndel@linux.intel.com>
Acked-by: Alan Cox <alan@redhat.com>
Acked-by: Jeff Garzik <jeff@garzik.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/serial/8250_pci.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/drivers/serial/8250_pci.c
+++ b/drivers/serial/8250_pci.c
@@ -1986,6 +1986,7 @@ static int pciserial_suspend_one(struct 
 
 static int pciserial_resume_one(struct pci_dev *dev)
 {
+	int err;
 	struct serial_private *priv = pci_get_drvdata(dev);
 
 	pci_set_power_state(dev, PCI_D0);
@@ -1995,7 +1996,9 @@ static int pciserial_resume_one(struct p
 		/*
 		 * The device may have been disabled.  Re-enable it.
 		 */
-		pci_enable_device(dev);
+		err = pci_enable_device(dev);
+		if (err)
+			return err;
 
 		pciserial_resume_ports(priv);
 	}
